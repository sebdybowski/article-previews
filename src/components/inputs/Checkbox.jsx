import React from 'react';
import PropTypes from 'prop-types';

export const Checkbox = ({ label, value, onValueChange }) => <div className="form-check">
    <input className="form-check-input" type="checkbox" id={label} onChange={() => onValueChange(label)} checked={value} />
    <label className="form-check-label" htmlFor={label}>
        <span className="btn-link">{label}</span>
    </label>
</div>;

Checkbox.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.bool,
    onValueChange: PropTypes.func.isRequired,
};