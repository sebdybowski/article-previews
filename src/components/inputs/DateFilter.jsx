import React from 'react';
import PropTypes from 'prop-types';

export const DateFilter = ({ onValueChange, sortOrder }) =>
    <button type="button" className="btn btn-link float-right" onClick={() => onValueChange()}>
        <span className={`sort-order ${sortOrder}`}>Sort by date</span>
    </button>;

DateFilter.propTypes = {
    onValueChange: PropTypes.func.isRequired,
    sortOrder: PropTypes.string,
};