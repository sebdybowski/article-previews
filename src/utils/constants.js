export const ENDPOINT_TYPE = {
    SPORTS: 'sports',
    FASHION: 'fashion',
};

export const CATEGORY_TYPE = ['sport', 'fashion'];

export const INITIAL_STATE = {
    articles: [],
    selectedCategories: CATEGORY_TYPE,
};