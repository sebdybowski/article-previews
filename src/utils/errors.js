import _ from 'lodash';

export const resolveErrorCategory = errorData => _.chain(errorData)
    .get('response.config.url')
    .split('/')
    .last()
    .value();