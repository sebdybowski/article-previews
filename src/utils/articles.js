import _ from 'lodash';
import {MONTH_MAP, SORT_ORDER} from "./constants";

export const resolveSortableDate = article => _.chain(article)
    .get('date')
    .split(' ')
    .map((value, index) => {
        if (index === 0 && value.length <= 2 ) return `0${value}`;
        if (index === 1) return _.get(MONTH_MAP, value);
        return value;
    })
    .reverse()
    .join('')
    .replace('.', '')
    .toNumber()
    .value();

export const resolveArticlesList = (articles, selectedCategories, sortOrder) => _.chain(articles)
    .filter(article => _.includes(selectedCategories, _.get(article, 'category')))
    .orderBy(['dateSortable'], [sortOrder])
    .value();
