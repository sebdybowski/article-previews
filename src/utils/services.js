import axios from 'axios';
import _ from 'lodash';
import { ENDPOINT_TYPE } from './constants';

const { SPORTS, FASHION } = ENDPOINT_TYPE;

export const fetchArticle = type => axios.get(`http://localhost:6010/articles/${type}`)
    .then(response => _.get(response, 'data.articles'))
    .catch(error => {
        console.log(error);
        return error;
    });

export const fetchAllArticles = () => Promise.all([fetchArticle(SPORTS), fetchArticle(FASHION)])
    .then(response => _.flatten(response))
    .catch(error => console.log(error));