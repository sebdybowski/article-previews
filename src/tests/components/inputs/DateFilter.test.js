import React from 'react';
import { shallow } from 'enzyme';
import { DateFilter } from '../../../components/inputs/DateFilter';
import { SORT_ORDER } from '../../../utils/constants';
import _ from 'lodash';

const mockProps = {
    onValueChange: _.noop,
};

describe('DateFilter', () => {
    it('should render correctly with sortOrder asc', () => {
        const component = shallow(<DateFilter {...mockProps} sortOrder={SORT_ORDER.ASCENDING} />);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly with sortOrder esc', () => {
        const component = shallow(<DateFilter {...mockProps} sortOrder={SORT_ORDER.DESCENDING} />);
        expect(component).toMatchSnapshot();
    });
});