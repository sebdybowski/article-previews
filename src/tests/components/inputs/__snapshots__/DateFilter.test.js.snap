// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFilter should render correctly with sortOrder asc 1`] = `
<button
  className="btn btn-link float-right"
  onClick={[Function]}
  type="button"
>
  <span
    className="sort-order asc"
  >
    Sort by date
  </span>
</button>
`;

exports[`DateFilter should render correctly with sortOrder esc 1`] = `
<button
  className="btn btn-link float-right"
  onClick={[Function]}
  type="button"
>
  <span
    className="sort-order desc"
  >
    Sort by date
  </span>
</button>
`;
